### ISSUE TITLE
Entity spawned in on_step not visible to `get_objects_inside_radius`

### ISSUE THREAD

**Issue by sfan5** [minetest/minetest#14614]
### Minetest version

```true
Minetest 5.9.0-dev-5ce59f7c5 (Linux)
```

### Operating system and version

Linux

### Summary

#13880 causes objects spawned during iteration of the object list (so during `on_step`) to only show up once this iteration ends.
This could be considered a defect. It's not easily fixable.

### Steps to reproduce

```lua
core.register_entity("test2:a", {
	initial_properties = {
		static_save = false,
		visual = "sprite",
		textures = {"cdb_update.png"},
	},
	on_step = function(self)
		local pos = self.object:get_pos()
		if not self.once then
			local obj = core.add_entity(pos, "test2:b")
			assert(obj ~= nil)
			for _, obj2 in ipairs(core.get_objects_inside_radius(pos, 1)) do
				if obj2:get_luaentity() and obj2:get_luaentity().name == "test2:b" then
					print("found it")
				end
			end
		end
		self.once = true
	end,
})
core.register_entity("test2:b", {
	initial_properties = {
		static_save = false,
		visual = "sprite",
		textures = {"cdb_add.png"},
	},
})
```

1. do `/spawnentity test2:a`
2. observe that both entities were spawned
3. look for "found it" line in vain

bonus:
4. change `on_step` to `on_activate`
5. re-do and observe that "found it" was printed

**Comment by nerzhul** [comment]
if i remember entities are activated asynchronously, when you are in on_step there are no activated yet, which require a full environment::step core side

**Comment by sfan5** [comment]
No, that's not it. Objects are directly activated here: https://github.com/minetest/minetest/blob/ae4cd1ebf18789b31cf01e465c0702b029c98d38/src/serverenvironment.cpp#L1894-L1896

It's due to the newly introduced data structure acting this way.

**Comment by appgurueu** [comment]
Removing the low priority label due to #15479.

It is true that when an entity is spawning entities inside `on_step`, it will often not have to "find them again" using `get_objects_inside_radius`. But as soon as third-party APIs like `tnt.boom` start being called in `on_step`, we want this to work.

**Comment by sofar** [comment]
This bug was fixed in `a3648b0b1` from #14631. It should be closed.

**Comment by sfan5** [comment]
Indeed.
